src/pybind/mgr/orchestrator.py

```diff
@@ -0,0 +1,456 @@
+
+        @return two-tuple of boolean, string
+        """
+        return None, None
+
+    def wait(self, completions):
+        """
+        Given a list of Completion instances, progress any which are
+        incomplete.  Return a true if everything is done.
+
+        Callers should inspect the detail of each completion to identify
+        partial completion/progress information, and present that information
+        to the user.
+
+        For fast operations (e.g. reading from a database), implementations
+        may choose to do blocking IO in this call.
+        """
+        raise NotImplementedError()
+
+    def get_inventory(self, node_filter=None):
+        """
+
+        :param node_filter:
+        :return: list of InventoryNode
+        """
+        raise NotImplementedError()
+
+    def describe_service(self, service_type, service_id):
+        """
+        Describe a service (of any kind) that is already configured in
+        the orchestrator.  For example, when viewing an OSD in the dashboard
+        we might like to also display information about the orchestrator's
+        view of the service (like the kubernetes pod ID).
+
+        When viewing a CephFS filesystem in the dashboard, we would use this
+        to display the pods being currently run for MDS daemons.
+        """
+        raise NotImplementedError()
+
+    def create_osds(self, osd_spec):
+        """
+        Create one or more OSDs within a single Drive Group.
+
+        The principal argument here is the drive_group member
+        of OsdSpec: other fields are advisory/extensible for any
+        finer-grained OSD feature enablement (choice of backing store,
+        compression/encryption, etc).
+
+        :param osd_spec: OsdCreationSpec
+        :return:
+        """
+        raise NotImplementedError()
+
+    def replace_osds(self, osd_spec):
+        """
+        Like create_osds, but the osd_id_claims must be fully
+        populated.
+        """
+        raise NotImplementedError()
+
+    def remove_osds(self, node, osd_ids):
+        """
+        :param node: A node name, must exist.
+        :param osd_ids: list of OSD IDs
+
+        Note that this can only remove OSDs that were successfully
+        created (i.e. got an OSD ID).
+        """
+        raise NotImplementedError()
+
+    def add_stateless_service(self, service_type, spec):
+        assert isinstance(spec, StatelessServiceSpec)
+        raise NotImplementedError()
+
+    def update_stateless_service(self, service_type, id_, spec):
+        assert isinstance(spec, StatelessServiceSpec)
+        raise NotImplementedError()
+
+    def remove_stateless_service(self, service_type, id_):
+        raise NotImplementedError()
+
+    def add_mon(self, node_name):
+        """
+        We operate on a node rather than a particular device: it is
+        assumed/expected that proper SSD storage is already available
+        and accessible in /var.
```

Comment:
minikube doesn't like `/var`. See [rook/rook/cluster/examples/kubernetes/ceph/cluster.yaml#L60](https://github.com/rook/rook/blob/master/cluster/examples/kubernetes/ceph/cluster.yaml#L60) and https://github.com/rook/rook/issues/1192